### ISSUE TITLE
Sqs GetQueueUrl does not set query params on request 

### ISSUE THREAD

**Issue by rbraeunlich** [aws/aws-sdk-java-v2#3972]
### Describe the bug

When debugging the creating of a `DefaultSdkHttpFullRequest` from a `GetQueueUrlRequest` I can see that no query parameters are being added to the request. My localstack then responds with 
"Missing Action in request for query-protocol service ServiceModel(sqs)"
because neither body nor request parameters contain an `Action`.

Using 2.20.59 the parameters are being added to the request.

### Expected Behavior

The query parameters should be added when creating the request.

### Current Behavior

```
10:02:49.938 32192 DEBUG lifecycle-thread-2 i.a.c.s.l.s.AbstractPollingMessageSource - Starting StandardSqsMessageSource for queue last-chance-expiring-formats-48h
10:02:55.661 37915 DEBUG lifecycle-thread-2 i.a.c.sqs.QueueAttributesResolver - Resolving attributes for queue last-chance-expiring-formats-48h
10:02:57.814 40068 DEBUG lifecycle-thread-2 s.a.a.c.i.ExecutionInterceptorChain - Creating an interceptor chain that will apply interceptors in the following order: [software.amazon.awssdk.core.internal.interceptor.HttpChecksumRequiredInterceptor@720122dd, software.amazon.awssdk.core.internal.interceptor.SyncHttpChecksumInTrailerInterceptor@60e770ad, software.amazon.awssdk.core.internal.interceptor.HttpChecksumValidationInterceptor@6df1f1ff, software.amazon.awssdk.core.internal.interceptor.AsyncRequestBodyHttpChecksumTrailerInterceptor@14785638, software.amazon.awssdk.core.internal.interceptor.HttpChecksumInHeaderInterceptor@6f1489a0, software.amazon.awssdk.awscore.interceptor.HelpfulUnknownHostExceptionInterceptor@342dfef7, software.amazon.awssdk.awscore.eventstream.EventStreamInitialRequestInterceptor@56b18f84, software.amazon.awssdk.awscore.interceptor.TraceIdExecutionInterceptor@aecafc, software.amazon.awssdk.services.sqs.endpoints.internal.SqsResolveEndpointInterceptor@4b3c1ac0, software.amazon.awssdk.services.sqs.endpoints.internal.SqsEndpointAuthSchemeInterceptor@10bd9f16, software.amazon.awssdk.services.sqs.endpoints.internal.SqsRequestSetEndpointInterceptor@4a2aef2a, software.amazon.awssdk.services.sqs.internal.MessageMD5ChecksumInterceptor@3ca525b9]
10:02:57.835 40089 DEBUG lifecycle-thread-2 s.a.a.c.i.ExecutionInterceptorChain - Interceptor 'software.amazon.awssdk.services.sqs.endpoints.internal.SqsEndpointAuthSchemeInterceptor@10bd9f16' modified the message with its modifyRequest method.
10:02:57.837 40091 TRACE lifecycle-thread-2 s.a.a.c.i.ExecutionInterceptorChain - Old: GetQueueUrlRequest(QueueName=last-chance-expiring-formats-48h)
New: GetQueueUrlRequest(QueueName=last-chance-expiring-formats-48h)
10:05:41.094 203348 DEBUG lifecycle-thread-2 s.a.a.c.i.ExecutionInterceptorChain - Interceptor 'software.amazon.awssdk.services.sqs.endpoints.internal.SqsRequestSetEndpointInterceptor@4a2aef2a' modified the message with its modifyHttpRequest method.
10:05:41.094 203348 TRACE lifecycle-thread-2 s.a.a.c.i.ExecutionInterceptorChain - Old: DefaultSdkHttpFullRequest(httpMethod=POST, protocol=http, host=127.0.0.1, port=55535, encodedPath=/, headers=[Content-Length, Content-Type, X-Amz-Target], queryParameters=[])
New: DefaultSdkHttpFullRequest(httpMethod=POST, protocol=http, host=127.0.0.1, port=55535, encodedPath=/, headers=[Content-Length, Content-Type, X-Amz-Target], queryParameters=[])
10:05:43.215 205469 DEBUG lifecycle-thread-2 software.amazon.awssdk.request - Sending Request: DefaultSdkHttpFullRequest(httpMethod=POST, protocol=http, host=127.0.0.1, port=55535, encodedPath=/, headers=[amz-sdk-invocation-id, Content-Length, Content-Type, User-Agent, X-Amz-Target], queryParameters=[])
10:05:43.429 205683 DEBUG lifecycle-thread-2 s.a.a.c.i.i.SdkLengthAwareInputStream - Specified InputStream length of 48 has been reached. Returning EOF.
10:05:43.650 205904 DEBUG lifecycle-thread-2 s.a.awssdk.auth.signer.Aws4Signer - AWS4 Canonical Request: POST
/

amz-sdk-invocation-id:a8571d9c-473d-5daf-a149-9219484f2933
amz-sdk-request:attempt=1; max=4
content-length:48
content-type:application/x-amz-json-1.0
host:127.0.0.1:55535
x-amz-date:20230505T080543Z
x-amz-target:AmazonSQS.GetQueueUrl

amz-sdk-invocation-id;amz-sdk-request;content-length;content-type;host;x-amz-date;x-amz-target
2a3bbe0a4dc25cd13f44bf5f09c1fb4f04eee00b46f42f491eb12bf29798f2c6
10:05:43.651 205905 DEBUG lifecycle-thread-2 s.a.awssdk.auth.signer.Aws4Signer - AWS4 String to sign: AWS4-HMAC-SHA256
20230505T080543Z
20230505/us-east-1/sqs/aws4_request
683bb9d6d38ca0567a5f10a9763c6b3b103fd953803a2d924d9308e576281b39
10:05:43.651 205905 TRACE lifecycle-thread-2 s.a.awssdk.auth.signer.Aws4Signer - Generating a new signing key as the signing key not available in the cache for the date: 1683273943427
10:05:45.371 207625 DEBUG lifecycle-thread-2 s.a.a.c.i.i.SdkLengthAwareInputStream - Specified InputStream length of 48 has been reached. Returning EOF.
10:05:46.254 208508 DEBUG aws-java-sdk-NettyEventLoop-2-2 software.amazon.awssdk.requestId - Received failed response: 500, Request ID: not available, Extended Request ID: not available
10:05:46.255 208509 DEBUG aws-java-sdk-NettyEventLoop-2-2 software.amazon.awssdk.request - Received failed response: 500, Request ID: not available, Extended Request ID: not available
10:05:46.267 208521 DEBUG aws-java-sdk-NettyEventLoop-2-2 s.a.a.h.n.n.i.n.HttpStreamsHandler - [Channel: [id: 0xb3b07053, L:/127.0.0.1:55601 - R:/127.0.0.1:55535]] Received LastHttpContent [id: 0xb3b07053, L:/127.0.0.1:55601 - R:/127.0.0.1:55535] with ignoreBodyRead as false
10:05:46.291 208545 DEBUG aws-java-sdk-NettyEventLoop-2-2 s.a.a.protocols.json.JsonContent - Unable to parse HTTP response content
java.io.UncheckedIOException: software.amazon.awssdk.thirdparty.jackson.core.JsonParseException: Unexpected character ('<' (code 60)): expected a valid value (JSON String, Number, Array, Object or token 'null', 'true' or 'false')
 at [Source: (byte[])"<?xml version='1.0' encoding='utf-8'?>
<ErrorResponse xmlns="http://queue.amazonaws.com/doc/2012-11-05/"><Error><Code>InternalError</Code><Message>exception while calling sqs with unknown operation: Operation detection failed. Missing Action in request for query-protocol service ServiceModel(sqs).</Message></Error><RequestId>2KKC1QFPW76G3FD5ZK9C5OUN45JQ8YRL6ZAEHZ37YKEOC64QM1LO</RequestId></ErrorResponse>"; line: 1, column: 2]
	at software.amazon.awssdk.utils.FunctionalUtils.asRuntimeException(FunctionalUtils.java:180) ~[utils-2.20.60.jar:na]
	at software.amazon.awssdk.utils.FunctionalUtils.lambda$safeSupplier$4(FunctionalUtils.java:110) ~[utils-2.20.60.jar:na]
	at software.amazon.awssdk.utils.FunctionalUtils.invokeSafely(FunctionalUtils.java:136) ~[utils-2.20.60.jar:na]
	at software.amazon.awssdk.protocols.jsoncore.JsonNodeParser.parse(JsonNodeParser.java:94) ~[json-utils-2.20.60.jar:na]
	at software.amazon.awssdk.protocols.json.JsonContent.parseJsonContent(JsonContent.java:74) ~[aws-json-protocol-2.20.60.jar:na]
	at software.amazon.awssdk.protocols.json.JsonContent.<init>(JsonContent.java:47) ~[aws-json-protocol-2.20.60.jar:na]
	at software.amazon.awssdk.protocols.json.JsonContent.createJsonContent(JsonContent.java:65) ~[aws-json-protocol-2.20.60.jar:na]
	at software.amazon.awssdk.protocols.json.internal.unmarshall.AwsJsonProtocolErrorUnmarshaller.unmarshall(AwsJsonProtocolErrorUnmarshaller.java:70) ~[aws-json-protocol-2.20.60.jar:na]
	at software.amazon.awssdk.protocols.json.internal.unmarshall.AwsJsonProtocolErrorUnmarshaller.handle(AwsJsonProtocolErrorUnmarshaller.java:66) ~[aws-json-protocol-2.20.60.jar:na]
	at software.amazon.awssdk.protocols.json.internal.unmarshall.AwsJsonProtocolErrorUnmarshaller.handle(AwsJsonProtocolErrorUnmarshaller.java:41) ~[aws-json-protocol-2.20.60.jar:na]
	at software.amazon.awssdk.core.http.MetricCollectingHttpResponseHandler.lambda$handle$0(MetricCollectingHttpResponseHandler.java:52) ~[sdk-core-2.20.60.jar:na]
	at software.amazon.awssdk.core.internal.util.MetricUtils.measureDurationUnsafe(MetricUtils.java:63) ~[sdk-core-2.20.60.jar:na]
	at software.amazon.awssdk.core.http.MetricCollectingHttpResponseHandler.handle(MetricCollectingHttpResponseHandler.java:52) ~[sdk-core-2.20.60.jar:na]
	at software.amazon.awssdk.core.internal.http.async.AsyncResponseHandler.lambda$prepare$0(AsyncResponseHandler.java:89) ~[sdk-core-2.20.60.jar:na]
	at java.base/java.util.concurrent.CompletableFuture$UniCompose.tryFire(CompletableFuture.java:1150) ~[na:na]
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:510) ~[na:na]
	at java.base/java.util.concurrent.CompletableFuture.complete(CompletableFuture.java:2147) ~[na:na]
	at software.amazon.awssdk.core.internal.http.async.AsyncResponseHandler$BaosSubscriber.onComplete(AsyncResponseHandler.java:132) ~[sdk-core-2.20.60.jar:na]
	at software.amazon.awssdk.http.nio.netty.internal.ResponseHandler$DataCountingPublisher$1.onComplete(ResponseHandler.java:515) ~[netty-nio-client-2.20.60.jar:na]
	at software.amazon.awssdk.http.nio.netty.internal.ResponseHandler.runAndLogError(ResponseHandler.java:250) ~[netty-nio-client-2.20.60.jar:na]
	at software.amazon.awssdk.http.nio.netty.internal.ResponseHandler.access$600(ResponseHandler.java:75) ~[netty-nio-client-2.20.60.jar:na]
	at software.amazon.awssdk.http.nio.netty.internal.ResponseHandler$PublisherAdapter$1.onComplete(ResponseHandler.java:371) ~[netty-nio-client-2.20.60.jar:na]
	at software.amazon.awssdk.http.nio.netty.internal.nrs.HandlerPublisher.publishMessage(HandlerPublisher.java:402) ~[netty-nio-client-2.20.60.jar:na]
	at software.amazon.awssdk.http.nio.netty.internal.nrs.HandlerPublisher.flushBuffer(HandlerPublisher.java:338) ~[netty-nio-client-2.20.60.jar:na]
	at software.amazon.awssdk.http.nio.netty.internal.nrs.HandlerPublisher.receivedDemand(HandlerPublisher.java:291) ~[netty-nio-client-2.20.60.jar:na]
	at software.amazon.awssdk.http.nio.netty.internal.nrs.HandlerPublisher.access$200(HandlerPublisher.java:61) ~[netty-nio-client-2.20.60.jar:na]
	at software.amazon.awssdk.http.nio.netty.internal.nrs.HandlerPublisher$ChannelSubscription$1.run(HandlerPublisher.java:495) ~[netty-nio-client-2.20.60.jar:na]
	at io.netty.util.concurrent.AbstractEventExecutor.runTask(AbstractEventExecutor.java:174) ~[netty-common-4.1.91.Final.jar:4.1.91.Final]
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:167) ~[netty-common-4.1.91.Final.jar:4.1.91.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:470) ~[netty-common-4.1.91.Final.jar:4.1.91.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:566) ~[netty-transport-4.1.91.Final.jar:4.1.91.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) ~[netty-common-4.1.91.Final.jar:4.1.91.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) ~[netty-common-4.1.91.Final.jar:4.1.91.Final]
	at java.base/java.lang.Thread.run(Thread.java:833) ~[na:na]
Caused by: software.amazon.awssdk.thirdparty.jackson.core.JsonParseException: Unexpected character ('<' (code 60)): expected a valid value (JSON String, Number, Array, Object or token 'null', 'true' or 'false')
 at [Source: (byte[])"<?xml version='1.0' encoding='utf-8'?>
<ErrorResponse xmlns="http://queue.amazonaws.com/doc/2012-11-05/"><Error><Code>InternalError</Code><Message>exception while calling sqs with unknown operation: Operation detection failed. Missing Action in request for query-protocol service ServiceModel(sqs).</Message></Error><RequestId>2KKC1QFPW76G3FD5ZK9C5OUN45JQ8YRL6ZAEHZ37YKEOC64QM1LO</RequestId></ErrorResponse>"; line: 1, column: 2]
	at software.amazon.awssdk.thirdparty.jackson.core.JsonParser._constructError(JsonParser.java:2391) ~[third-party-jackson-core-2.20.60.jar:na]
	at software.amazon.awssdk.thirdparty.jackson.core.base.ParserMinimalBase._reportError(ParserMinimalBase.java:735) ~[third-party-jackson-core-2.20.60.jar:na]
	at software.amazon.awssdk.thirdparty.jackson.core.base.ParserMinimalBase._reportUnexpectedChar(ParserMinimalBase.java:659) ~[third-party-jackson-core-2.20.60.jar:na]
	at software.amazon.awssdk.thirdparty.jackson.core.json.UTF8StreamJsonParser._handleUnexpectedValue(UTF8StreamJsonParser.java:2737) ~[third-party-jackson-core-2.20.60.jar:na]
	at software.amazon.awssdk.thirdparty.jackson.core.json.UTF8StreamJsonParser._nextTokenNotInObject(UTF8StreamJsonParser.java:902) ~[third-party-jackson-core-2.20.60.jar:na]
	at software.amazon.awssdk.thirdparty.jackson.core.json.UTF8StreamJsonParser.nextToken(UTF8StreamJsonParser.java:794) ~[third-party-jackson-core-2.20.60.jar:na]
	at software.amazon.awssdk.protocols.jsoncore.JsonNodeParser.parse(JsonNodeParser.java:116) ~[json-utils-2.20.60.jar:na]
	at software.amazon.awssdk.protocols.jsoncore.JsonNodeParser.lambda$parse$1(JsonNodeParser.java:97) ~[json-utils-2.20.60.jar:na]
	at software.amazon.awssdk.utils.FunctionalUtils.lambda$safeSupplier$4(FunctionalUtils.java:108) ~[utils-2.20.60.jar:na]
	... 32 common frames omitted
10:05:49.696 211950 TRACE sdk-async-response-1-0 s.a.a.c.r.c.TokenBucketRetryCondition - Successfully acquired token bucket capacity to retry this request. Acquired: 5. Remaining: 495
10:05:49.697 211951 DEBUG sdk-async-response-1-0 software.amazon.awssdk.request - Retryable error detected. Will retry in 80ms. Request attempt number 2
```

### Reproduction Steps

```kotlin
package some.package

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.BeforeAll
import org.junit.jupiter.api.Test
import org.testcontainers.containers.localstack.LocalStackContainer
import org.testcontainers.junit.jupiter.Container
import org.testcontainers.junit.jupiter.Testcontainers
import org.testcontainers.utility.DockerImageName
import software.amazon.awssdk.auth.credentials.AwsBasicCredentials
import software.amazon.awssdk.auth.credentials.StaticCredentialsProvider
import software.amazon.awssdk.regions.Region
import software.amazon.awssdk.services.sqs.SqsAsyncClient

@Testcontainers
class FooTest {

    companion object {
        @Container
        val localstack: LocalStackContainer =
            LocalStackContainer(DockerImageName.parse("localstack/localstack:2.0.2"))

        @JvmStatic
        @BeforeAll
        fun setUp() {
            localstack.execInContainer(
                "awslocal",
                "sqs",
                "create-queue",
                "--queue-name",
                "my-queue",
            )
        }
    }

    @Test
    fun foo() {
        val sqsAsyncClient = SqsAsyncClient.builder()
            .endpointOverride(localstack.getEndpointOverride(LocalStackContainer.Service.SQS))
            .region(Region.of(localstack.region))
            .credentialsProvider(
                StaticCredentialsProvider.create(
                    AwsBasicCredentials.create(
                        localstack.accessKey,
                        localstack.secretKey
                    )
                )
            ).build()

        val queueUrl = sqsAsyncClient.getQueueUrl { it.queueName("my-queue") }.get().queueUrl()

        assertThat(queueUrl).isNotNull()
    }
}
```

### AWS Java SDK version used

2.20.60

### JDK version used

openjdk-17

### Operating System and version

Win 10

**Comment by daviddenton** [comment]
Have been digging into this because it has broken our SQS fakes - and have found out that this is because of the switch to AWS JSON-1.0 protocol in the client instead of the URL encoded form parameters.

Wondering if there are other changes coming which will also break other services coming up?

**Comment by kellertk** [comment]
Hi there. @daviddenton is correct, this is because the SQS service has changed the wire protocol format for the API. The service itself will respond to both formats to support old and new API clients, but new versions of the SDK will use the new JSON API instead of the older calling semantics. Third party mock libraries (and possibly localstack) will need to be updated to expect the new calling conventions. See also boto/botocore#2930.

**Comment by kellertk** [comment]
Please note: changes to SQS were reverted in 2.20.61.

**Comment by daviddenton** [comment]
Is there a timescale for this change being re-released? We need to spend time updating our fake SQS implementation so would appreciate the heads up of how long we have to fix it up. 🙃. (https://github.com/http4k/http4k-connect/issues/242)

**Comment by debora-ito** [comment]
The SQS protocol change was released again, and is included in version `2.21.19`.

Note that a partial change was released in `2.21.18` but a bug was found in the SQS model and fixed in `2.21.19`. Make sure you are using the version with the fix.

If you find any issues with the SQS protocol change, let us know.

**Comment by debora-ito** [comment]
Are you sending the request to the SQS endpoint directly, or are you using Localstack (or any other third-party solution)?

https://github.com/aws/aws-sdk-java/issues/3055
